### PR TITLE
ovirt: use go-ovirt ConflictErr instead of string check

### PR DESCRIPTION
**Pull request by bennyz**

### CODE REVIEW

**Review by ahadas — APPROVED** [review]
@bennyz verified?

File: pkg/controller/plan/adapter/ovirt/client.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (r *Client) CreateSnapshot(vmRef ref.Ref) (snapshot string, err error) {
 			MustBuild(),
 	).Query("correlation_id", correlationID).Send()
 	if err != nil {
-		if strings.Contains(err.Error(), "Cannot create Snapshot") {
+		if errors.Is(err, &ovirtsdk.ConflictError{}) {
```
Comment (bennyz) [inline pkg/controller/plan/adapter/ovirt/client.go:68]: changing to errors.As

**Comment by bennyz** [comment]
@ahadas let's hold off on this, verification failed

**Comment by bennyz** [comment]
So it seems the conflict is handled fine, but looks like CDI doesn't really like unexpected snapshots in the chain:
```
E0810 10:42:00.192334 1 data-processor.go:278] snapshot backing file '1e010d72-a1cb-4fe8-be6f-7da76fec225a' does not match expected checkpoint '25353462-f0cc-425d-b6b8-379ef0153976', unable to safely rebase snapshot
kubevirt.io/containerized-data-importer/pkg/importer.(*ImageioDataSource).Transfer
/remote-source/app/pkg/importer/imageio-datasource.go:152
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).initDefaultPhases.func2
/remote-source/app/pkg/importer/data-processor.go:208
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
/remote-source/app/pkg/importer/data-processor.go:275
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessData
/remote-source/app/pkg/importer/data-processor.go:184
main.handleImport
```

**Comment by ahadas** [comment]
@bennyz and that happens when we fail to remove one snapshot and proceed with the other? (That shouldn't happen, no?)

**Comment by bennyz** [comment]
> @bennyz and that happens when we fail to remove one snapshot and proceed with the other? (That shouldn't happen, no?)

this happened when I created a snapshot right before forklift

**Comment by ahadas** [comment]
Ok, that really doesn't sound related to our recent changes then

**Comment by ahadas** [comment]
we agreed offline to bump the dependencies first and get rid of our fork of go-ovirt

**Review by ahadas — APPROVED** [review]

**Pull request merged**